app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -43,6 +43,11 @@ def custom_index_relation(collection)
       param :checksum_type, String, :desc => N_("Checksum of the repository, currently 'sha1' & 'sha256' are supported")
       param :docker_upstream_name, String, :desc => N_("Name of the upstream docker repository")
       param :docker_tags_whitelist, Array, :desc => N_("Comma-separated list of tags to sync for Container Image repository")
+      RepositoryTypeManager.repository_types.each do |_, type|
```

Comment:
The params aren't showing up under `"repository"` because they aren't being wrapped above at this line:
```ruby
wrap_parameters :repository, :include => RootRepository.attribute_names.concat([:ignore_global_proxy])
```

So, we need to replace that line above with something like the following:
```ruby
    generic_repo_wrap_params = []
    RepositoryTypeManager.repository_types.each do |_, type|
      type.repository_remote_options.each do |option|
        generic_repo_wrap_params << option.name
      end
    end
    repo_wrap_params = RootRepository.attribute_names.concat([:ignore_global_proxy]) + generic_repo_wrap_params

    wrap_parameters :repository, :include => repo_wrap_params
```